projects/igniteui-angular/src/lib/grids/filtering/excel-style/excel-style-search.component.ts

```diff
@@ -152,17 +201,26 @@ export class IgxExcelStyleSearchComponent implements AfterViewInit, OnDestroy {
             });
         });
         esf.columnChange.pipe(takeUntil(this.destroy$)).subscribe(() => {
-            this.virtDir.resetScrollPosition();
+            this.virtDir?.resetScrollPosition();
         });
 
         esf.listDataLoaded.pipe(takeUntil(this.destroy$)).subscribe(() => {
+            if (this.isHierarchical() && this.esf.listData[0].isSpecial) {
```

Comment:
I would suggest to remove the "Select all" and "Add to current..." from the list data to avoid checking the isHierarchical flag at multiple places. We could create these two items outside the igxFor of the list similarly to what we have for the tree component.

Reply:
We have agreed to leave it like this for now.